### ISSUE TITLE
TypeVar-parameterized descriptor incorrectly flagged as asymmetric

### ISSUE THREAD

**Issue by saryou** [microsoft/pyright#9450]
**Overview**
A recent change in Pyright (starting from version 1.1.387) appears to have introduced a bug in the handling of descriptor asymmetry. When a descriptor is parameterized with TypeVar, it is incorrectly flagged as asymmetric, even if the getter and setter types match, leading to inaccurate type inference.

**Steps to Reproduce**
The following code demonstrates incorrect type inference with a TypeVar-parameterized descriptor.

```python
from typing import Any, overload, Self


class GenericDescriptor[GT, ST]:
    @overload
    def __get__(self, instance: None, owner: Any) -> Self:
        ...

    @overload
    def __get__(self, instance: Any, owner: Any) -> GT:
        ...

    def __set__(self, instance: Any, value: ST):
        ...


class NonGenericDescriptor:
    @overload
    def __get__(self, instance: None, owner: Any) -> Self:
        ...

    @overload
    def __get__(self, instance: Any, owner: Any) -> int | None:
        ...

    def __set__(self, instance: Any, value: int | None):
        ...


class Object:
    generic = GenericDescriptor[int | None, int | None]()
    non_generic = NonGenericDescriptor()


obj = Object()
obj.generic = 1
obj.non_generic = 1
reveal_type(obj.generic)  # int | None which seems wrong
reveal_type(obj.non_generic)  # Literal[1] as expected

**Comment by erictraut** [comment]
Thanks for the bug report.

This is actually a long-standing bug, but it was previously masked. A recent change uncovered it.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.389
